drivers/disk/flashdisk.c

```diff
@@ -28,13 +28,13 @@ struct flashdisk_data {
 	struct disk_info info;
 	struct k_mutex lock;
 	const unsigned int area_id;
-	const off_t offset;
+	const k_off_t offset;
```

Comment:
I am not sure whether using `off_t` or `k_off_t` is correct here.
I guess that the `off_t` has been used here because it may have larger range than `size_t`, but using signed int here is pointless, same with `cached_addr`. Note also that using 64bit values here, just to serve device that is >2GB and <=4GB, means that upper 4 bytes of the `off_t` will never be used.

I think that we need separate types for storage/external devices, with long addresses, independent from whatever kernel is running on.

Reply:
`k_off_t` (and `off_t`) are specifically for describing sizes and offsets within storage devices.

Either is fine with me, but this PR is specifically addressing the layering violation.

It might be better to retype this field in a separate PR.